code/modules/mob/living/carbon/xenomorph/castes/afflictor/abilities_afflictor.dm

```diff
@@ -0,0 +1,249 @@
+	if(!.)
+		return
+	if(get_dist(owner, A) > 1)
+		to_chat(owner, "<span class='warning'>We need to be next to our prey.</span>")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/reagent_slash/proc/figure_out_living_target(atom/A) //aim assist
+	var/clickDir = get_dir(owner, A)
+	var/turf/presumedPos = get_step(owner, clickDir)
+	var/mob/living/L = locate() in presumedPos
+	if((presumedPos == get_turf(A) && L == null) || L == owner || istype(L, /mob/living/carbon/xenomorph))
+		to_chat(owner, "<span class='warning'>Our slash won't affect this target!</span>")
+		return A
+	return L
+
+/datum/action/xeno_action/activable/reagent_slash/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/atom/T
+	if(isturf(A))
+		T = A
+	else
+		T = get_turf(A)
+	var/atom/hold = figure_out_living_target(T,	owner)
+	var/atom/Z = hold
+	if(X.is_ventcrawling)
+		return
+	if(isnull(Z))
+		return
+	if(isturf(Z))
+		return
+	if(!Z?.can_sting())
+		to_chat(owner, "<span class='warning'>Our slash won't affect this target!</span>")
+		return FALSE
+	var/slash_count = 4
+	var/reagent_transfer_amount = 4
+	succeed_activate()
+	slash_action(Z, X.selected_reagent, channel_time = 1.2 SECONDS, count = slash_count, transfer_amount = reagent_transfer_amount)
+	add_cooldown()
+
+/datum/action/xeno_action/activable/reagent_slash/proc/slash_action(mob/living/carbon/C, toxin = /datum/reagent/toxin/xeno_neurotoxin, channel_time = 1 SECONDS, transfer_amount = 4, count = 3)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/limb/affecting = X.zone_selected
+	if(!C?.can_sting())
+		return FALSE
+	var/datum/reagent/body_tox
+	var/i = 1
+	SEND_SIGNAL(X, COMSIG_HIVE_XENO_REAGENT_SLASH)
+	do
+		X.face_atom(C)
+		if(X.stagger)
+			return FALSE
+		body_tox = C.reagents.get_reagent(toxin)
+		C.reagents.add_reagent(toxin, transfer_amount)
+		if(!body_tox) //Let's check this each time because depending on the metabolization rate it can disappear between stings.
+			body_tox = C.reagents.get_reagent(toxin)
+		if(get_dist(X, C) > 1)
+			to_chat(X, "<span class='warning'>We need to be closer to [C].</span>")
+			return
+		playsound(C, "alien_claw_flesh", 25, TRUE)
+		playsound(C, 'sound/effects/spray3.ogg', 15, TRUE)
+		if(C.getToxLoss() > 6 && toxin == /datum/reagent/toxin/xeno_praelyx)
+			C.apply_damage(damage = 10, damagetype = BRUTE, def_zone = affecting, sharp = TRUE)
+			C.adjustToxLoss(-6)
+		C.attack_alien_harm(X, dam_bonus = 0, set_location = affecting, random_location = FALSE, no_head = FALSE, no_crit = FALSE, force_intent = null)
+		X.visible_message(C, "<span class='danger'>The [X] swipes at [C]!</span>")
+		X.do_attack_animation(C)
+	while(i++ < count && do_after(X, channel_time, TRUE, C, BUSY_ICON_HOSTILE, ignore_turf_checks = TRUE, target_can_move = TRUE))
+	return TRUE
+
+
+// ***************************************
+// *********** NANOCRYSTAL CAMOUFLAGE
+// ***************************************
+/datum/action/xeno_action/xeno_camouflage
```

Comment:
this looks like a copy paste of hunter stealth but not, why is this intruding on hunter niche?